jenkins/PerfCI_Grafana_Deployment/Jenkinsfile

```diff
@@ -69,74 +69,91 @@ pipeline {
-                                PYTHONPATH=. sudo python3 benchmark_runner/grafana/update_grafana_dashboard.py --grafana_url="${GRAFANA_URL}" --grafana_api_key="${GRAFANA_API_KEY}" --grafana_json_path="${GRAFANA_JSON_PATH}" --grafana_folder_name="${GRAFANA_FOLDER_NAME}";
-                                cp -p benchmark_runner/grafana/perf/dashboard_backup.json benchmark_runner/grafana/perf/dashboard.json;
-                            '''
-                            sh 'git add benchmark_runner/grafana/perf/dashboard.json'
-                        }
 
-                        // Update main.libsonnet
-                        if (CHANGES_MAIN_LIBSONNET.toBoolean()) {
+        stage('Update Grafana files and commit changes') {
+            steps {
+                script {
+                    try {
+                        dir(WORKSPACE) {
+                            // Git configuration
                             sh '''
-                                cp -p benchmark_runner/grafana/perf/jsonnet/main.libsonnet benchmark_runner/grafana/perf/jsonnet/main.libsonnet_backup
-                                # Add code to update main.libsonnet here
+                                git checkout main
+                                git config --global user.email "${EMAIL}"
+                                git config --global user.name "${USER_NAME}"
+                                git config pull.rebase false  # merge
                             '''
-                            sh 'git add benchmark_runner/grafana/perf/jsonnet/main.libsonnet'
-                        }
 
-                        // Check if there are changes to commit
-                        def hasChanges = sh(script: 'git diff --exit-code', returnStatus: true) != 0
-
-                        if (hasChanges) {
-                            // Commit, pull, and push
-                            sh '''
-                                git commit -m "Update Grafana files"
-                                git pull https://${GIT_TOKEN}@${GIT_REPOSITORY} main
-                                git push https://${GIT_TOKEN}@${GIT_REPOSITORY} main
-                            '''
-                        } else {
-                            echo 'No changes to commit.'
+                            // Update dashboard.json
+                            if (CHANGES_DASHBOARD.toBoolean()) {
+                                sh '''
+                                    cp -p benchmark_runner/grafana/perf/dashboard.json benchmark_runner/grafana/perf/dashboard_backup.json;
+                                    PYTHONPATH=. sudo python3 benchmark_runner/grafana/update_grafana_dashboard.py --grafana_url="${GRAFANA_URL}" --grafana_api_key="${GRAFANA_API_KEY}" --grafana_json_path="${GRAFANA_JSON_PATH}" --grafana_folder_name="${GRAFANA_FOLDER_NAME}";
+                                    cp -p benchmark_runner/grafana/perf/dashboard_backup.json benchmark_runner/grafana/perf/dashboard.json;
```

Comment:
This looks like it's simply undoing any change to dashboard.json created by update_grafana_dashboard.py?

Reply:
update_grafana_dashboard.py, update for new versions in dashboard.json file

Reply:
Why are you backing up dashboard.json, running update_grafana_dashboard.py, and then restoring dashboard.json from the backup, presumably overwriting whatever update_grafana_dashboard.py changes?